File: 01 Array/Contains Duplicate/Duplicates.java
import java.util.HashMap;
import java.util.Scanner;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashMap<Integer, Integer> counter = new HashMap<Integer, Integer>();
        for (int x : nums) {
            if (counter.getOrDefault(x, 0) >= 1) {
                return true;
            }
            counter.put(x, counter.getOrDefault(x, 0) + 1);
        }
        return false;
    }
}

public class Duplicates {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t > 0) {
            int n = sc.nextInt();
            int[] nums = new int[n];
            for (int i = 0; i < n; i++) {
                nums[i] = sc.nextInt();
            }
            Solution obj = new Solution();
            System.out.println(obj.containsDuplicate(nums));
            t--;
        }
        sc.close();
    }
}

File: 01 Array/Contains Duplicate/Duplicates.go
package main

import "fmt"

func duplicates(nums []int) bool {
	counter := make(map[int]int)
	for _, x := range nums {
		if counter[x] >= 1 {
			return true
		}
		counter[x]++
	}
	return false
}

func main() {
	var t int
	fmt.Scan(&t)
	for t > 0 {
		t--
		var n int
		fmt.Scan(&n)
		nums := make([]int, n)
		for i := 0; i < n; i++ {
			fmt.Scan(&nums[i])
		}
		fmt.Println(duplicates(nums))
	}
}
